http/notification_endpoint.go

```diff
@@ -403,6 +414,134 @@ func decodePatchNotificationEndpointRequest(ctx context.Context, r *http.Request
 	return req, nil
 }
 
+// handlePostNotificationEndpointTest is the HTTP handler for the PUT /api/v2/notificationEndpoints/:id/test route.
+func (h *NotificationEndpointHandler) handlePutNotificationEndpointTest(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	edp, err := decodePutNotificationEndpointRequest(ctx, r)
+	if err != nil {
+		h.Logger.Debug("failed to decode request", zap.Error(err))
+		h.HandleHTTPError(ctx, err, w)
+		return
+	}
+
+	auth, err := pctx.GetAuthorizer(ctx)
+	if err != nil {
+		h.HandleHTTPError(ctx, err, w)
+		return
+	}
+
+	var token *influxdb.Authorization
```

Comment:
This looks too hacky. @influxdata/compute-team has cleaned this up already for task. We should make sure the query proxy server can handle the context.

Reply:
@kelwang this is what is currently done in the `/api/v2/query` endpoint. I agree it's hacky and that we need a long term solution, but I'm not sure exactly what that looks like.